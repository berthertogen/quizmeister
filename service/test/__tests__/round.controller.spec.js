let questionIds;

test("should create round on Post", async () => {
  const responseQuestion = await buildPostRequest("question", "", questionFactory.create());
  questionIds = [responseQuestion.body.questionId];
  const response = await buildPostRequest("round", "", roundFactory.create(questionIds));
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(roundFactory.ignoreProperties(true));
});

test("should get round on Get", async () => {
  const response = await buildGetRequest('round', '1');
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(roundFactory.ignoreProperties(true));
});

test("should get round on Get (search)", async () => {
  const response = await buildGetRequest('round', 'search/0/20');
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot([roundFactory.ignoreProperties(true)]);
});

test("should get themes on Get (themes)", async () => {
  const response = await buildGetRequest('round', 'themes');
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot();
});

test("should update round on Put", async () => {
  const response = await buildPutRequest("round", "1", roundFactory.update(questionIds));
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(roundFactory.ignoreProperties(true));
});

test("should copy round on Put (copy)", async () => {
  const response = await buildPutRequest("round", "1/copy", []);
  assertContentTypeJsonAndStatus200(response);
  expect(response.body).toMatchSnapshot(roundFactory.ignoreProperties(false));
});

test("should delete round on Delete", async () => {
  const responseDelete1 = await buildDeleteRequest("round", "1/true");
  assertContentTypeJsonAndStatus200(responseDelete1);
  const responseDelete2 = await buildDeleteRequest("round", "2/true");
  assertContentTypeJsonAndStatus200(responseDelete2);
  const responseGetRounds = await buildGetRequest('round', 'search/0/20');
  expect(responseGetRounds.status).toBe(200);
  expect(responseGetRounds.body).toMatchSnapshot();
  const responseGetQuestions = await buildGetRequest('question', 'search/0/20');
  expect(responseGetQuestions.status).toBe(200);
  expect(responseGetQuestions.body).toMatchSnapshot();
});
