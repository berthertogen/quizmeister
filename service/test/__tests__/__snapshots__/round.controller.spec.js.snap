// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should copy round on Put (copy) 1`] = `
Object {
  "modifiedOn": Any<String>,
  "questions": Array [],
  "remark": "remark",
  "roundId": 2,
  "searchField": Any<String>,
  "shortId": Any<String>,
  "theme": "theme",
  "title": "updated title - copy",
}
`;

exports[`should create round on Post 1`] = `
Object {
  "modifiedOn": Any<String>,
  "questions": Array [
    Object {
      "answers": null,
      "modifiedOn": Any<String>,
      "questionId": 1,
      "remark": "remark",
      "rounds": Array [
        Object {
          "modifiedOn": Any<String>,
          "questions": null,
          "remark": "remark",
          "roundId": 1,
          "searchField": Any<String>,
          "shortId": Any<String>,
          "theme": "theme",
          "title": "title",
        },
      ],
      "scoring": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "title": "title",
      "type": 2,
    },
  ],
  "remark": "remark",
  "roundId": 1,
  "searchField": Any<String>,
  "shortId": Any<String>,
  "theme": "theme",
  "title": "title",
}
`;

exports[`should delete round on Delete 1`] = `Array []`;

exports[`should delete round on Delete 2`] = `Array []`;

exports[`should get round on Get (quiz) 1`] = `
Array [
  Object {
    "modifiedOn": Any<String>,
    "questions": Array [
      Object {
        "answers": Array [
          Object {
            "answerId": 1,
            "correct": true,
            "searchField": "trueanswer 1",
            "text": "answer 1",
          },
          Object {
            "answerId": 2,
            "correct": false,
            "searchField": "falseanswer 2",
            "text": "answer 2",
          },
        ],
        "modifiedOn": Any<String>,
        "questionId": 1,
        "remark": "remark",
        "rounds": Array [
          Object {
            "modifiedOn": Any<String>,
            "questions": null,
            "remark": "remark",
            "roundId": 1,
            "searchField": Any<String>,
            "shortId": Any<String>,
            "theme": "theme",
            "title": "title",
          },
        ],
        "scoring": Object {
          "scoringId": 1,
          "timeLimitSeconds": 0,
          "timeScoringInterval": 0,
          "type": 1,
          "weightCorrectAnswer": 1,
          "weightNoAnswer": 0,
        },
        "searchField": Any<String>,
        "shortId": Any<String>,
        "title": "title",
        "type": 2,
      },
    ],
    "remark": "remark",
    "roundId": 1,
    "searchField": Any<String>,
    "shortId": Any<String>,
    "theme": "theme",
    "title": "title",
  },
]
`;

exports[`should get round on Get (search) 1`] = `
Array [
  Object {
    "modifiedOn": Any<String>,
    "questions": Array [
      Object {
        "answers": null,
        "modifiedOn": Any<String>,
        "questionId": 1,
        "remark": "remark",
        "rounds": Array [
          Object {
            "modifiedOn": Any<String>,
            "questions": null,
            "remark": "remark",
            "roundId": 1,
            "searchField": Any<String>,
            "shortId": Any<String>,
            "theme": "theme",
            "title": "title",
          },
        ],
        "scoring": null,
        "searchField": Any<String>,
        "shortId": Any<String>,
        "title": "title",
        "type": 2,
      },
    ],
    "remark": "remark",
    "roundId": 1,
    "searchField": Any<String>,
    "shortId": Any<String>,
    "theme": "theme",
    "title": "title",
  },
]
`;

exports[`should get round on Get 1`] = `
Object {
  "modifiedOn": Any<String>,
  "questions": Array [
    Object {
      "answers": null,
      "modifiedOn": Any<String>,
      "questionId": 1,
      "remark": "remark",
      "rounds": Array [
        Object {
          "modifiedOn": Any<String>,
          "questions": null,
          "remark": "remark",
          "roundId": 1,
          "searchField": Any<String>,
          "shortId": Any<String>,
          "theme": "theme",
          "title": "title",
        },
      ],
      "scoring": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "title": "title",
      "type": 2,
    },
  ],
  "remark": "remark",
  "roundId": 1,
  "searchField": Any<String>,
  "shortId": Any<String>,
  "theme": "theme",
  "title": "title",
}
`;

exports[`should get themes on Get (themes) 1`] = `
Array [
  "theme",
]
`;

exports[`should update round on Put 1`] = `
Object {
  "modifiedOn": Any<String>,
  "questions": Array [
    Object {
      "answers": null,
      "modifiedOn": Any<String>,
      "questionId": 1,
      "remark": "remark",
      "rounds": Array [
        Object {
          "modifiedOn": Any<String>,
          "questions": null,
          "remark": "remark",
          "roundId": 1,
          "searchField": Any<String>,
          "shortId": Any<String>,
          "theme": "theme",
          "title": "updated title",
        },
      ],
      "scoring": null,
      "searchField": Any<String>,
      "shortId": Any<String>,
      "title": "title",
      "type": 2,
    },
  ],
  "remark": "remark",
  "roundId": 1,
  "searchField": Any<String>,
  "shortId": Any<String>,
  "theme": "theme",
  "title": "updated title",
}
`;
